### PR TITLE
Date truncate function

### PR DESCRIPTION
**Pull request by samansmink**
Implements the date_trunc function from ticket #193. Currently only supports Date and Datetime types. According to postgres documentation the date_trunc function should support Interval and Time as well but since Interval type is not implemented yet and Time type should be cast to Interval by this function those types are not supported yet.

### CODE REVIEW

File: src/function/scalar/date/date_trunc.cpp [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+#include "duckdb/common/string_util.hpp"
+
+// TODO date_trunc function should also handle interval data type when it is implemented. See
+// https://www.postgresql.org/docs/9.1/functions-datetime.html
+
+using namespace std;
+
+namespace duckdb {
+
+DateTruncSpecifier GetDateTruncSpecifier(string specifier) {
```
Comment (Mytherin) [inline src/function/scalar/date/date_trunc.cpp]: It seems like this is mostly shared with date_part and the DatePartSpecifier, perhaps we could merge them together?

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! It looks good. I will just request one change: merge the DateTruncSpecifier with the DatePartSpecifier. It seems like they have a lot of overlap and I would prefer not ot have this duplicate code. An exception can be thrown if date_trunc is called with an unsupported specifier (e.g. epoch or dow). Otherwise it looks great!

File: src/function/scalar/date/date_trunc.cpp (hunk 1 above)
Comment (samansmink) [reply src/function/scalar/date/date_trunc.cpp]: Sure!

**Comment by samansmink** [comment]
refactored it. I have a question though, now GetDatePartSpecifier is declared in date_part_specifier.hpp and defined in date_part.cpp but also used in date_trunc.cpp. Is that okay or should I move the definition somewhere else to make more clear that it is not only used by date_part.cpp?

**Comment by Mytherin** [comment]
The way it is is fine. Thanks for the effort! I will merge the commit now.

**Pull request merged**